tests/test_paired_inventory_accounting.py

```
@@ -0,0 +1,103 @@
+    assert "pair inverted" in why
+
+
+def test_proportional_merge_cost_basis_math(tmp_path):
+    """Partial merge calculates cost basis proportionally without phantom loss."""
+    db_file = tmp_path / "test_orders.db"
+    reg = OrderRegistry(db_path=db_file)
+    cond = "0xcond123"
+
+    # Insert historical orders and fills: 20 UP shares @ 0.30 ($6.00), 5 DOWN shares @ 0.65 ($3.25)
+    # Total historical spend = $9.25
+    o1 = OrderRecord(id="u1", condition_id=cond, token_id="tok_up", side="BUY", price=0.30, original_size=20, status="filled", posted_ts=1, last_polled_ts=1, run_id="r1")
```

Ruff (0.16.2)

[error] Possible hardcoded password assigned to argument: "token_id"

(S106)